lib/TutorialServo/TutorialServo.cpp

```diff
@@ -0,0 +1,50 @@
+
+#include "TutorialServo.h"
+
+TutorialServo::TutorialServo(PinName servoPin, float servoRangeInDegrees, float minPulsewidthInMs,
+                             float maxPulsewidthInMs)
+    : m_servoPwmOut(servoPin),
+      m_servoRangeInDegrees(servoRangeInDegrees),
+      m_minPulsewidthInMs(minPulsewidthInMs),
+      m_maxPulsewidthInMs(maxPulsewidthInMs) {
+  m_servoPwmOut.period(0.020);
+  // setting the member variables using initializer list
+  // otherwise nothing to do
+}
+
+TutorialServo::~TutorialServo() {
+  m_servoRangeInDegrees = 0;
```

Comment:
you should not need a destructor unless you are playing with raw pointers. The default destructor in this case should suffice.

Reply:
should i remove the destructor entirely or leave it as a function that does nothing?

Reply:
I'd say remove it, generally, you don't want functions that do nothing and is never used to stay in source files

Reply:
sounds good thanks